atlp-rwanda/rca-barefoot-team4-bn
# #184638235 Feat: Logout Functionality

<b>What does this PR do?</b>
This PR adds a logout endpoint to the backend API that allows users to securely log out of their accounts and end their active session.

<b>Description of Task to be completed?</b>

- Create a new endpoint /logout that handles POST requests
- Implement the logic to invalidate the user's session token and clear their session data from the server
- Return a response with a status code indicating success or failure

<b>How should this be manually tested?</b>

- Log in to the application
- Send a POST request to /logout
- Verify that the user's session token is invalidated and their session data is cleared from the server
- Attempt to access a protected endpoint by using the user's session token
- Verify that the request is rejected with a status code indicating unauthorized access

<b>Any background context you want to provide?</b>
Previously, users could not log out of their accounts, which posed a security risk as anyone with access to the user's device could potentially access their account.
